services/addons/images/firmware_manager/upgrader.py

```diff
@@ -71,6 +72,44 @@ def wait_until_online(self):
             iter += 1
         # 3 minutes pass with no response
         return -1
+    
+    def check_online(self):
+        iter = 0
+        # Ping once every second for 5 seconds to verify RSU is online
+        while iter < 5:
+            code = subprocess.run(
+                ["ping", "-n", "-c1", self.rsu_ip], capture_output=True
+            ).returncode
+            if code == 0:
+                return 0
+            iter += 1
+            time.sleep(1)
+        # 5 seconds pass with no response
+        return -1

```

Comment:
When I initially read this this, my first thought was that the 'check_online' function was returning the number of RSUs that were online. I see now that it returns 0 upon successful ping. I think the clarity of this code could be improved by returning True upon success and False upon failure.

Reply:
Updated this to return true/false for clarity